hm_1/tasks/task_01_05.py

```diff
@@ -0,0 +1,20 @@
+"""
+Given a list of integers numbers "nums".
+You need to find a sub-array with length less equal to "k", with maximal sum.
```

Comment:
> sub-array with length less equal to "k"

Did you implement this requirement?

Reply:
Yes, I implement this in the last condition, in "return" I don't consider cases where k>length(nums)  => we consider cases only where k<=length(nums).

Reply:
> sub-array with length less equal to "k"

This means that you need to find a subarray and this subarray can be smaller when k. Consider the following example:  `find_maximal_subarray_sum([4, -1, -1], 3) == 4`  subbary of the lenhgt 1 `[4]` has max sum.

Reply:
I don't quite understand, because find_maximal_subarray_sum([4, -1, -1], 3) == 4 must be equal 2, because size of subarray (k) = size of list and we need to return a sum of its elements, when find_maximal_subarray_sum([4, -1, -1], 1) == 4, or find_maximal_subarray_sum([4], 1) == 4  It must be so or I'm not right?

Reply:
We need to check all possible subarrays with sizes from 1 to k. 

```
# 3
[4, -1, -1]
# 2
[4, -1]
[-1, -1]
# 1
[4]
[-1]
[-1]
```